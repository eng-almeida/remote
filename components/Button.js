import styled from 'styled-components';

const Button = styled.button`
  display: inline-block;
  text-align: center;
  background: ${({ theme, variant = 'primary' }) => variant === 'secondary' ? 'transparent' : theme.colors.purple[200] };
  border: 2px solid ${({ theme }) => theme.colors.purple[200] };
  border-radius: ${({ theme }) => theme.radii.large};
  box-shadow: ${({ theme }) => theme.shadows[0]};
  padding: ${({ theme }) => `${theme.space[3]} ${theme.space[6]}`};
  color: ${({ theme, variant = 'primary' }) => variant === 'secondary' ? theme.colors.purple[200] : theme.colors.white };
  font-size: 16px;
  line-height: 19px;
  font-family: inherit;
  min-width: 142px;
`
export default Button;