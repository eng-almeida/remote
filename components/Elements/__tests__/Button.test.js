import Button from '@/components/Elements/Button';

describe('Button component', () => {
	it.each([['primary'], ['secondary']])('should match %s snapshot', variant => {
		const { container } = renderWithProviders(<Button variant={variant}>Test</Button>);
		expect(container).toMatchSnapshot();
	});
})