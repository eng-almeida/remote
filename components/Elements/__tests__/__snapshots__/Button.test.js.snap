// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match primary snapshot 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  background: #624DE3;
  border: 2px solid #624DE3;
  border-radius: 50px;
  box-shadow: 0px 6px 12px rgba(98,77,227,0.3);
  padding: 12px 26px;
  color: #ffffff;
  font-size: 16px;
  line-height: 19px;
  font-family: inherit;
  min-width: 142px;
}

.c0 > svg {
  margin-right: 8px;
}

<div>
  <button
    class="c0"
  >
    Test
  </button>
</div>
`;

exports[`Button component should match secondary snapshot 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  background: transparent;
  border: 2px solid #624DE3;
  border-radius: 50px;
  box-shadow: 0px 6px 12px rgba(98,77,227,0.3);
  padding: 12px 26px;
  color: #624DE3;
  font-size: 16px;
  line-height: 19px;
  font-family: inherit;
  min-width: 142px;
}

.c0 > svg {
  margin-right: 8px;
}

<div>
  <button
    class="c0"
  >
    Test
  </button>
</div>
`;
